sklearn/metrics/_pairwise_distances_reduction/_middle_term_computer.pyx.tp

```diff
@@ -417,8 +350,12 @@ cdef class SparseSparseMiddleTermComputer{{name_suffix}}(MiddleTermComputer{{nam
 
     def __init__(
         self,
-        X,
-        Y,
+        const DTYPE_t[:] X_data,
+        const SPARSE_INDEX_TYPE_t[:] X_indices,
+        const SPARSE_INDEX_TYPE_t[:] X_indptr,
+        const DTYPE_t[:] Y_data,
```

Comment:
Thanks, this is now fixed.

This one is interesting, though, because in the main branch we currently cast `X_data` and `Y_data` to float64, and we call the routine `sparse_sparse_middle_term_computation_64` for both `SparseMiddleTermComputer{32, 64}`.

I saw that you changed this behavior in your work on `SparseDenseDatasetsPair` to avoid casting. Could you provide more details about this decision?